src/lib/tooltip/tooltip.ts

```diff
@@ -261,7 +262,7 @@ export class MatTooltip implements OnDestroy {
     this._tooltipInstance = overlayRef.attach(portal).instance;
 
     // Dispose of the tooltip when the overlay is detached.
-    overlayRef.detachments().subscribe(() => {
+    merge(this._tooltipInstance!.afterHidden(), overlayRef.detachments()).subscribe(() => {
```

Comment:
Move the `!` up to the assignment for `tooltipInstance`?

Reply:
TS isn't smart enough to figure it out when it's being assigned to a property, it only works for variables.